orc/src/main/java/org/apache/iceberg/orc/OrcToIcebergVisitor.java

```diff
@@ -43,7 +43,12 @@ public Optional<Types.NestedField> record(
                 .map(Optional::get)
                 .collect(Collectors.toList()));
     return Optional.of(
-        Types.NestedField.of(icebergIdOpt.get(), isOptional, currentFieldName(), structType));
+        Types.NestedField.builder()
```

Comment:
did you mean to add a `builder()` method? Seems like this is currently missing

Reply:
Oops. That was done in #12211. This one depends on that going in first.